library/core/class.model.php

```diff
@@ -967,17 +967,22 @@ public static function editContentTimeout($data, &$timeLeft = 0) {
      *
      * @param string $lockKey Cache key to be assigned.
      * @param int $gracePeriod Period of time the key will stay valid.
-     * @return bool $hasMasterKey Whether a master key has been assigned.
+     * @return bool Whether a master key has been assigned.
      */
-    protected static function buildCacheLock(string $lockKey, int $gracePeriod = 60):bool {
-        $hasMasterKey = null;
+    protected static function buildCacheLock(string $lockKey, int $gracePeriod = 60): bool {
+        static $hasMasterKey = null;
+        $masterKey = null;
 
         if (is_null($hasMasterKey)) {
             // First instance will receive the $master key and lock the record.
             $instanceKey = getmypid();
-            $masterKey = Gdn::cache()->add($lockKey, $instanceKey, [
+            $keyAdded = Gdn::cache()->add($lockKey, $instanceKey, [
                 Gdn_Cache::FEATURE_EXPIRY => $gracePeriod
             ]);
+
+            if ($keyAdded) {
+                $masterKey = Gdn::cache()->get($lockKey);
```

Comment:
If the key was successfully added, meaning it didn't already exist, then we already know we have the current "master key", right?